lib/qiita/markdown/filters/external_link.rb

```diff
@@ -0,0 +1,35 @@
+module Qiita
+  module Markdown
+    module Filters
+      class ExternalLink < HTML::Pipeline::Filter
+        def call
+          return doc unless hostname
+
+          doc.search("a").each do |anchor|
+            next if anchor["href"].nil?
+            href = anchor["href"].strip
+            href_host = host_of(href)
+            next unless href_host
+            if href_host == hostname || !href_host.end_with?(".#{hostname}")
+              anchor["rel"] = "nofollow"
+              anchor["target"] = "_blank"
+            end
+          end
+
+          doc
+        end
+
+        private
+
+        def host_of(url)
+          m = url.match(%r{^https?://(.+?):?(\d+)?(/.*)?$})
+          m[1] if m
+        end
```

Comment:
@yujinakayama If the passed URL includes multibyte character, the `parse` method will throw `InvalidURIError`. Thus, I used the regular expression...